mapswipe_workers/mapswipe_workers/firebase_to_postgres/transfer_results.py

```diff
@@ -55,24 +42,92 @@ def transfer(current_results):
                 f"We will not transfer these"
             )
             continue
+        else:
+            transfer_results_for_project(project_id)
+
+    return project_id_list
+
+
+def transfer_results_for_project(project_id):
+    """Transfer the results for a specific project.
+
+    Download all results for a project from firebase.
+    Save results into an in-memory file.
+    Copy the results to postgres.
+    Delete results in firebase.
+
+    We are NOT using a Firebase transaction functions here anymore.
+    This has caused problems, in situations where a lot of mappers are
+    uploading results to Firebase at the same time. Basically, this is
+    due to the behaviour of Firebase Transaction function:
+
+        "If another client writes to this location
```

Comment:
Can I suggest adding a link to the source? This page is not what you quoted, but also has helpful info on transactions https://firebase.google.com/docs/firestore/manage-data/transactions#transaction_failure

Reply:
I added a link to the Firebase python sdk where I copied the statement from.

You link refers to FireStore, which is another thing. :)